File: App/Resources/AlgorithmDetails/stoogesort/stoogesort.swift
func stoogeSort(_ arr: inout [Int],
_ i: Int,
_ j: Int) {
    if arr[i] > arr[j] {
        arr.swapAt(i, j)
    }
    if j - i > 1 {
        let t = (j - i + 1) / 3
        stoogeSort(&arr, i, j - t)
        stoogeSort(&arr, i + t, j)
        stoogeSort(&arr, i, j - t)
    }
}

func sort(_ array: inout [Int]) {
    stoogeSort(&array, 0, array.count - 1)
}

var array: [Int] = [
    0, 39, 21, 62, 91, 77, 14, 23,
    90, 69, 51, 81, 68, 83, 32, 56,
]
sort(&array)
print(array)

File: App/Resources/AlgorithmDetails/stoogesort/stoogesort.rb
def stooge_sort(arr, i, j)
  if arr[j] < arr[i]
    arr[i], arr[j] = arr[j], arr[i]
  end
  if j - i > 1
    t = (j - i + 1) / 3
    stooge_sort(arr, i, j - t)
    stooge_sort(arr, i + t, j)
    stooge_sort(arr, i, j - t)
  end
end

def sort(array)
  stooge_sort(array, 0, array.length - 1)
end

array = [0, 39, 21, 62, 91, 77, 14, 23,
  90, 69, 51, 81, 68, 83, 32, 56]
sort(array)
p array
